Cubical/Codata/M/AsLimit/stream.agda

```diff
@@ -24,7 +29,167 @@ cons : ∀ {A} -> A -> stream A -> stream A
 cons x xs = in-fun (x , λ { tt -> xs } )
 
 hd : ∀ {A} -> stream A -> A
-hd {A} S = out-fun S .fst
+hd {A} s = out-fun s .fst
 
 tl : ∀ {A} -> stream A -> stream A
-tl {A} S = out-fun S .snd tt
+tl {A} s = out-fun s .snd tt
+
+------------------------------
+-- Equality of stream types --
+------------------------------
+
+open Stream
+
+stream-to-Stream : ∀ {A : Set} → stream A → Stream A
+head (stream-to-Stream x) = hd x
+tail (stream-to-Stream x) = stream-to-Stream (tl x)
+
+private
+  Stream-to-stream-func-x : ∀ {A : Set} (n : ℕ) -> Stream A → Wₙ (stream-S A) n
+  Stream-to-stream-func-x 0 x = lift tt
+  Stream-to-stream-func-x (suc n) x = head x , λ _ → Stream-to-stream-func-x n (tail x)
+
+  Stream-to-stream-func-π : ∀ {A : Set} (n : ℕ) (x : Stream A) → πₙ (stream-S A) (Stream-to-stream-func-x (suc n) x) ≡ Stream-to-stream-func-x n x
+  Stream-to-stream-func-π 0 x = refl {x = lift tt}
+  Stream-to-stream-func-π (suc n) x = λ i → head x , λ _ → Stream-to-stream-func-π n (tail x) i
+
+Stream-to-stream : ∀ {A : Set} -> Stream A -> stream A
+Stream-to-stream s = lift-to-M Stream-to-stream-func-x Stream-to-stream-func-π s
+
+hd-to-head : ∀ {A : Set} (b : Stream A) → hd (Stream-to-stream b) ≡ head b
+hd-to-head {A = A} b = refl
+
+head-to-hd : ∀ {A : Set} (b : stream A) → head (stream-to-Stream b) ≡ hd b
+head-to-hd {A = A} b = refl
+
+tail-to-tl : ∀ {A : Set} (b : stream A) → tail (stream-to-Stream b) ≡ stream-to-Stream (tl b)
+tail-to-tl b = refl
+
+private
+  tail-eq-x : ∀ {A : Set} → (b : Stream A) (n : ℕ) → fst (tl (Stream-to-stream b)) n ≡ fst (Stream-to-stream (tail b)) n
+  tail-eq-x {A = A} b n =
+    transport (λ i → (Wₙ (stream-S A) n)) (Stream-to-stream-func-x n (tail b))
+      ≡⟨ sym (transport-filler (λ i → Wₙ (stream-S A) n) (Stream-to-stream-func-x n (tail b))) ⟩
+    fst (Stream-to-stream (tail b)) n ∎
+
+  postulate
```

Comment:
Why postulate?

Reply:
I have not been able to complete the proof.